src/System.Private.Uri/tests/FunctionalTests/UriTests.cs

```diff
@@ -731,13 +734,16 @@ public static void TestIsWellFormedOriginalString()
             b = uri.IsWellFormedOriginalString();
             Assert.False(b);
 
-            uri = new Uri(@"c:\\directory\filename");
-            b = uri.IsWellFormedOriginalString();
-            Assert.False(b);
+            if (s_IsWindowsSystem)
+            {
+                uri = new Uri(@"c:\\directory\filename");
+                b = uri.IsWellFormedOriginalString();
+                Assert.False(b);
 
-            uri = new Uri(@"file://c:/directory/filename");
-            b = uri.IsWellFormedOriginalString();
-            Assert.False(b);
+                uri = new Uri(@"file://c:/directory/filename");
+                b = uri.IsWellFormedOriginalString();
+                Assert.False(b);
+            }
```

Comment:
The tests verify these are invalid Uris on Unix (https://github.com/dotnet/corefx/pull/15925/files#diff-37ac644755d2501e671ac255288654a3R1295) The constructor throws for these strings.
```
+                // Implicit Windows drive 
+                yield return new object[] { @"c:\", UriKind.Absolute };
+                yield return new object[] { @"c:\path\file", UriKind.Absolute };
+                
+                // Implicit UNC 
+                yield return new object[] { @"\\unchost", UriKind.Absolute };
+                yield return new object[] { @"\\unchost\", UriKind.Absolute };
+                yield return new object[] { @"\\unchost\path\file", UriKind.Absolute };
```

> I also wonder whether we can/should try to make the implementation recognize, on unix, unix file paths in addition to rather than instead of windows ones.

Technically it is possible. I doubt whether it will be used.